### ISSUE TITLE
Remove the requirement for authors to provide dimensions for the map element?

### ISSUE THREAD

**Issue by Malvoz** [Maps4HTML/HTML-Map-Element#31]
As I understand it, there is a requirement for authors to provide the dimensions for the map element (with no description for how UAs should handle cases when they're missing):

> The required `width` and `height` attributes of the `map` element allow the HTML author to explicitly set the size of the rectangular area devoted to the map. If the attributes are not set directly by the HTML author, their values must be established via CSS width and height properties or through script. In any case, these values must be established in order for the `map` to process and display its content. The map can be re-sized via changes to those properties.

In HTML dimension attributes "may" be specified: https://html.spec.whatwg.org/#dimension-attributes.

A few SVG elements had such a requirement, [that requirement was dropped](https://www.w3.org/TR/SVG2/geometry.html#Sizing) in SVG2.

Instead, I think we can define a [default object size](https://www.w3.org/TR/css-images-3/#default-object-size) (this is 300x150px for iframes), which I think would better match the expectations from authors.

**Comment by prushforth** [comment]
Yes the width and height attributes are redundant if you can specify those properties with CSS, and having two ways to accomplish one goal is unwanted complexity.  I was not aware of the default size for iframes.

I've found that Leaflet, which I use in the polyfill, is very unhappy when the height of the element you use is 0 (which seems to happen pre-layout but during custom element construction), but having two sources of width/height info has been confusing to say the least, and that confusion is probably reflected in the custom element code.

**Comment by zcorpan** [comment]
I agree that `width` and `height` should probably be optional and default to 300x150.

This could either be with a default aspect ratio of 2:1 where if you set one of width or height, the other changes to maintain the aspect ratio (like `video`), or independent defaults of 300 and 150 (like `canvas`).

**Comment by prushforth** [comment]
Update: I guess we should update the spec to reflect this change.

**Comment by prushforth** [comment]
@malvoz how should the author set the size of the map with CSS?

I have been using this to set the map to the available size, but I am getting a 300x150 map now:

```
         <style>
           html {height: 100%} body,map {height: inherit} * {margin: 0;padding: 0;}
        </style>
```

**Comment by Malvoz** [comment]
Where's the above code in use? We deleted that custom CSS (map element inheriting height) in https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/108.

You can set the dimensions like so:
```CSS
map {
  width: 100%;
  height: 100%;
}
```

or in pixels, or other units. The idea is also that the author should be able to set the `width` and `height` HTML attributes (values in CSS pixels), on the `<map>` element (where CSS takes precedence). But I don't think we're doing that yet with `attributeChangedCallback`(?).

**Comment by prushforth** [comment]
It's in a branch that we're working on.  I admit it is a convoluted way to set a map size, your suggestion works.  Thanks

**Comment by Malvoz** [comment]
@prushforth you could uncomment/re-use [these](https://github.com/Maps4HTML/Web-Map-Custom-Element/blob/5f4ebde5b7d2e82fa8468687def65c35ac5b2de4/index-web-map.html#L22-L32) demo styles to override the new default CSS to get the "full viewport" look.

**Comment by prushforth** [comment]
Yes I saw that just now and did exactly that.  This is way more direct and I like it.  Thanks.
